apps/backoffice-v2/src/common/components/molecules/DateRangePicker/DateRangePicker.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { useState, ComponentProps, useEffect } from 'react';
+import { CalendarIcon } from '@radix-ui/react-icons';
+import { formatDate } from '@/common/utils/format-date';
+import { ctw } from '@/common/utils/ctw/ctw';
+import { Button } from '../../atoms/Button/Button';
+import { Calendar } from '../../organisms/Calendar/Calendar';
+import { Popover, PopoverContent, PopoverTrigger } from '@ballerine/ui';
+import { DateRange } from 'react-day-picker';
+
+type TDateRangePickerProps = {
+  onChange: ComponentProps<typeof Calendar>['onSelect'];
+  value: ComponentProps<typeof Calendar>['value'];
+  className: ComponentProps<'div'>['className'];
+};
+
+export const DateRangePicker = ({ onChange, value, className }: TDateRangePickerProps) => {
+  const [date, setDate] = useState<DateRange | undefined>();
```

Comment:
Why are we not using `value` over this state?

Reply:
The `value` should update since `onChange` is passed. `onChange` should not be optional. If `value` was used as initial I'd expect it to be passed into the `useState`. 

If we were to make `value` optional (not saying we should) it would be fine to pass it into the `useState` as well since its already initialized with `undefined`.

 I don't understand why we are passing `value` and making it required in the interface but it doesn't do anything. I do see now its used for a conditional class.

If `value` is from the URL and `onChange` updates the URL I'm failing to understand why it wouldn't work. 🙂

Reply:
Yes, I completely understand and agree with you, the issue with the `value`  is that it remains undefined even after selecting the date range, due to which `{!value?.from && !value?.to && <span>Pick a date</span>}` always remains active even after selecting the date range.

![Screenshot (332)](https://github.com/ballerine-io/ballerine/assets/140178357/d005eb1b-1cc0-47c7-88a8-152d74580e80)


And as the type of `onChange`  is `SelectRangeEventHandler | undefined` here, we cannot invoke an object that is possibly `undefined` thus need to use optional chaining.

Reply:
Addressed in https://github.com/ballerine-io/ballerine/pull/2328/commits/38644b7d689519a1a247c0fb28b27e67a6ce187a